pkg/rollout/rollout.go

```diff
@@ -318,6 +364,23 @@ func (r *Rollout) diagnoseCandidate(candidate string, healthCriteria []config.Me
 	return d, errors.Wrap(err, "failed to diagnose candidate's health")
 }
 
+// shouldRollForward determines if enough time has passed since last roll
+// forward.
+func (r *Rollout) shouldRollForward(lastRollForwardStr string, timeBetweenRollForwards int) (bool, error) {
+	lastRollForward, err := time.Parse(time.RFC3339, lastRollForwardStr)
```

Comment:
if annotation is empty, return a clear error saying X annotation is missing

Reply:
I added a TODO for this